### PR TITLE
Added method to handle incoming objects

### PR DESCRIPTION
**Pull request by delvh**
(not finished as the methods to further process these "Events" have not
been implemented yet)

### CODE REVIEW

File: src/main/java/envoy/server/ObjectMessageProcessor.java [hunk 1]
```diff
@@ -18,14 +24,63 @@
  */
 public class ObjectMessageProcessor implements IMessageProcessor {
 
+	private long currentUserID = 0;// temporary
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/server/ObjectMessageProcessor.java]: Please mark this clearly as a TODO item.

File: src/main/java/envoy/server/ObjectMessageProcessor.java [hunk 2]
```diff
@@ -18,14 +24,63 @@
...
+	 * @param writeProxy the writeProxy to define the resulting socket for
+	 * @param obj        the object that has been read out of the {@link Message}
+	 * @throws IllegalArgumentException if the object given is neither an
+	 *                                  {@link envoy.data.Message}, an {@link Event}
+	 *                                  nor a {@link LoginCredentials}
+	 * @since Envoy Server Standalone v0.1-alpha
+	 */
+	private void handleObject(Message request, WriteProxy writeProxy, Object obj) throws IllegalArgumentException {
+		boolean	responseToSameSocket	= false;
+		Object	usage;
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/server/ObjectMessageProcessor.java]: As `usage` has the type `Object` casting its value to other types does not have any effect. However, the cast object could be directly passed into a handler specific to that type.

File: src/main/java/envoy/server/ObjectMessageProcessor.java [hunk 3]
```diff
@@ -18,14 +24,63 @@
...
+			usage = cast.getClass();
+		} else if (obj instanceof Event) {// if object is Event
+			usage = (Event<?>) obj;
+		} else if (obj instanceof LoginCredentials) {// if object is LoginCredential
+			responseToSameSocket = true;
+			LoginCredentials cast = (LoginCredentials) obj;
+			usage = new User(currentUserID++, cast.getName());
+		} else throw new IllegalArgumentException();
+
+		Message response = writeProxy.getMessage();
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/server/ObjectMessageProcessor.java]: A response to the client's request is not always necessary and sometimes delayed. We will probably handle server to client communication in a separate thread.

File: src/main/java/envoy/server/ObjectMessageProcessor.java (hunk 2 above)
Comment (delvh) [reply src/main/java/envoy/server/ObjectMessageProcessor.java]: I don't get what I should do to improve the code

**Pull request merged**